addons/mail/models/mail_mail.py

```diff
@@ -477,6 +480,31 @@ def _send(self, auto_commit=False, raise_exception=False, smtp_session=None):
                     except Exception:
                         pass
 
+                # Compute email size and transform attachment in downloadable link in the body if it exceeds the limit
+                estimated_email_size = (
+                        (len(json.dumps(headers, ensure_ascii=False).encode(errors='ignore')) +
+                         len(body.encode(errors='ignore')) +
+                         sum(attachment['file_size'] for attachment in
+                             attachments.sudo().read(['file_size']))) * BASE_64_SIZE_OVERHEAD_RATIO +
+                        EMAIL_SIZE_SECURITY_MARGIN_BYTES
+                )
+
+                attachments_links = None
+                if attachments and estimated_email_size > max_email_size_in_bytes:
+                    # Remove attachments and prepare downloadable link to be added in the body
+                    for attachment in attachments:
+                        attachment.generate_access_token()
+                    attachments_links = self.env['ir.qweb']._render('mail.mail_attachment_links',
+                                                                    {'attachments': attachments})
+                    attachments = self.env['ir.attachment']
```

Comment:
Also a good point, thanks. Mails sent (mail_mail) from the chatter with attachment are marked auto_delete  and then to_delete. They are then processed by the cron “Mail: Email Queue Manager” which delete the mail_mail but left the attachment untouched as it is linked to mail_message. So it seems safe but the comment of the method _postprocess_sent_message seems to says the contrary (“including deleting it completely along with its attachment if the ``auto_delete`` flag of the mail was set.”). What do you think @tde-banana-odoo?

Reply:
Mmmh, depends probably of the use case
  * comment -> we delete the mail but keep the mail_message and attachments -> should work
  * mass mailing -> we delete the mail and the underlying mail_message -> annoying

Note that we delete attachments linked to the "mail_message", not the attachments linked to the record itself (see the override of unlink in MailMessage). Classic chatter usage links attachments to the record itself so it should not be an issue in that case (probably to check a bit further to be sure).

When doing a pure mass mailing, indeed we might have an issue, could be great to check the composer in mass mailing mode and mailing application, to see what happens with attachments (model / res_id, as I don't remember all details). Then we can check how to avoid the issue.